### PR TITLE
feat: send red packet json to native on mobile platforms

### PR DESCRIPTION
**Pull request by jk234ert**
## Description

Closes # (NO_ISSUE)

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

**Comment by nuanyang233** [comment]
Please merge develop.

File: packages/mask/src/plugins/RedPacket/SNSAdaptor/RedPacket/index.tsx [hunk 1]
```diff
@@ -134,6 +135,21 @@ export function RedPacket(props: RedPacketProps) {
...
+            availability: {
+                token_address: availability.token_address,
+                balance: availability.balance,
+                total: availability.total,
+                claimed: availability.claimed,
+                expired: availability.expired
+            },
+            postLink: postLink.toString()
+        })
+    }, [availability, payload, postLink])
```
Comment (hanszhoou) [inline packages/mask/src/plugins/RedPacket/SNSAdaptor/RedPacket/index.tsx]:
```suggestion
    }, [availability, JSON.stringify(payload), postLink])
```

Comment (jk234ert) [reply packages/mask/src/plugins/RedPacket/SNSAdaptor/RedPacket/index.tsx]: Leave `payload` as a raw object is fine, the native side could easily decode and map it as a JSON object.
With `stringify`, the native side still has to `un-stringify` the payload.

Comment (UncleBill) [reply packages/mask/src/plugins/RedPacket/SNSAdaptor/RedPacket/index.tsx]: `[availability, payload, postLink]` here is a dependency array that `useCallback` uses to compare between two renderings to check if needs to return a new function to assign to `onCalimOrRefundOnNative` here, it's about React not business layer, somehow performance improvement.

In JS, two objects with same shape could referrer to different objects, but serialization of them with `stringify` will be the same. 

Here, if the payload passed from parent component has a same shape among every rendering, and other entities of the dependency array have the same value, `useCallback` will return the exactly same `onClaimOrRefundOnNative`.

**Review by Jack-Works — DISMISSED** [review]

File: packages/plugins/FileService/src/SNSAdaptor/MainDialog.tsx [hunk 2]
```diff
@@ -75,7 +75,7 @@ const FileServiceDialog: React.FC<Props> = (props) => {
             open={props.open}
             title={t.__display_name()}
             onClose={onDecline}>
-            <DialogContent style={{ minWidth: 515 }}>
+            <DialogContent>
```
Comment (nuanyang233) [inline packages/plugins/FileService/src/SNSAdaptor/MainDialog.tsx]: It may cause pc side style bug.

Comment (yanzhihong23) [reply packages/plugins/FileService/src/SNSAdaptor/MainDialog.tsx]: Tested on both pc and mobile.

**Review by Jack-Works — DISMISSED** [review]

**Review by Jack-Works — DISMISSED** [review]

**Comment by wenluomask** [comment]
- [x] 1. Cannot decrypt old backup with wallet, popup will show "Failed to encrypt secret";

Screen Shot 2022-02-24 at 5 45 50 PM

Screen Shot 2022-02-24 at 5 54 49 PM


- [x] 2. For backup without wallet, it actually decrypted successfully, but it show the failure message;

Screen Shot 2022-02-24 at 5 48 06 PM

Screen Shot 2022-02-24 at 5 49 43 PM

- [x] 3. Contacts in different planforms (Twitter, Ins and Facebook) will all showed in same tab;

Screen Shot 2022-02-24 at 5 59 42 PM

- [x] 4. Contacts were exactly same for any planform;

https://user-images.githubusercontent.com/85149605/155502734-dbd28113-625e-46a0-aba6-0ea322b4d2aa.mov

**Comment by wenluomask** [comment]
5. Currently, I can't get any new contacts, even from Twitter;
For the Facebook, Ins and Minds, the contact lists are empty.

**Review by Jack-Works — DISMISSED** [review]

**Comment by wenluomask** [comment]
Still failed to backup or recover contacts.

I added some contacts and then did the local back up, but I can't get the same contact list after recover this backup file.

File: packages/mask/src/setup.ui.1.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import Services from './extension/service'
```
Comment (Jack-Works) [inline packages/mask/src/setup.ui.1.ts]: why your PR contains this file? it hasn't been merged to the main branch yet...

Comment (jk234ert) [reply packages/mask/src/setup.ui.1.ts]: Only for test

**Review by Jack-Works — DISMISSED** [review]

File: packages/shared-base/src/Messages/Mask.ts [hunk 4]
```diff
@@ -61,7 +61,8 @@ export interface MaskEvents extends MaskSettingsEvents, MaskMobileOnlyEvents, Ma
     restoreSuccess: void
     relationsChanged: RelationChangedEvent[]
     pluginMinimalModeChanged: [id: string, newStatus: boolean]
-
+    compositionAttachMetaData: [metaId: string, meta?: unknown]
+    compositionDropMetaData: [metaId: string]
```
Comment (Jack-Works) [inline packages/shared-base/src/Messages/Mask.ts:65]:
```suggestion
    app_compositionAttachMetaData: [metaId: string, meta?: unknown]
    app_compositionDropMetaData: [metaId: string]
```

File: packages/public-api/src/mobile.ts [hunk 5]
```diff
@@ -112,6 +114,20 @@ export interface SharedNativeAPIs {
         pageOption?: PageOption
     }): Promise<MobilePostRecord[]>
     update_post(params: { post: Partial<MobilePostRecord>; options: { mode: 0 | 1 } }): Promise<MobilePostRecord[]>
+    /**
+     * Mask Plugins
+     */
+    notifyRedpacket(params: {
+        redpacketPayload: any
```
Comment (Jack-Works) [inline packages/public-api/src/mobile.ts:121]: Shouldn't it be `RedPacketAvailability`?

File: packages/mask/background/tasks/NotificationsToMobile.ts [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import { notify } from 'async-call-rpc/full'
```
Comment (Jack-Works) [inline packages/mask/background/tasks/NotificationsToMobile.ts]: This file already appears in `packages/mask/background/tasks/NotificationsToMobile.ts`. Please update with develop branch.

**Pull request closed without merging**